### DISCUSSION TITLE
sorting and writing a parquet file seems very slow ?

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4381] [Q&A] [answered]
I am using DuckDB and pyarrow to write Hive table

```
import duckdb 
import pyarrow as pa
import pyarrow.dataset as ds
con = duckdb.connect()
result = con.execute( '''
         select *, extract ( year from l_shipdate) as year
         from 'C:/xxxxx/parquet/lineitem.parquet')
          order by l_shipdate
         ''').fetch_record_batch()

pa.dataset.write_dataset(result,
                        base_dir="/Users/mimoune.djouallah/Desktop/TPC-H-SF10/parquet/lineitem",
                        format ='parquet',
                        partitioning=['year'],
                        partitioning_flavor="hive",
                        existing_data_behavior='overwrite_or_ignore')
```

the file is located here  : https://drive.google.com/file/d/18gv0Yd_a-Zc7CSolol8qeYVAAzSthnSN/view?usp=sharing

it take 10 minute to finish, is this expected ?

**Comment by Mytherin** [comment]
There were some improvements made to the arrow conversion recently, can you retry this with the latest master?

**Comment by djouallah** [comment]
I am using DuckDB master

**Reply by Mytherin** [reply]
Are you sure the problem is on the DuckDB side, and not on the Arrow writing side?

**Comment by djouallah** [comment]
I don't know :)  but because duckDB don't write partitions I have to use Pyarrow:)  hint, hint 😜😜

**Comment by djouallah** [comment, marked as answer]
@Mytherin find the issue, when I remove order by, it works nearly instantly !!! so I guess it is something with DuckDB

**Reply by Mytherin** [reply]
DuckDB has a highly optimized sort implementation, but despite that sorting is still expensive. How much RAM does your pc have? If the data set does not fit in memory and data has to be offloaded to disk it can slow down quite significantly. How much time does it take if you reduce the data set size to 1GB?
